### PR TITLE
Use common Fiware-Correlator header in all the platform components

### PR DESCRIPTION
**Pull request by gtorodelvalle**
- Fixes https://github.com/telefonicaid/context-adapter/issues/78
- No linting issues
- 100% tests passed

### CODE REVIEW

File: lib/context_adapter_communication.js [hunk 1]
```diff
@@ -32,21 +32,22 @@ var caLogger = require('logops'),
...
   var requestOptions = {
     method: 'POST',
     uri: 'http://' + caConfig.CB_HOST + ':' + caConfig.CB_PORT + caConfig.CB_PATH + '/queryContext',
     headers: {
       'Accept': 'application/json',
       'Content-Type': 'application/json',
-      'Fiware-Service': service || caConfig.SERVICE,
-      'Fiware-ServicePath': servicePath || caConfig.SERVICE_PATH
+      'Unica-Correlator': caHelper.getUnicaCorrelator(request),
+      'Authorization': caHelper.getTransactionId(request),
```
Comment (fgalan) [inline lib/context_adapter_communication.js]: As far as I understand, the mechanism to transport transaction ID is yet under discussion. Thus, I'd recomment to put this PR on hold while the mechanism gets clarified.

File: lib/context_adapter_communication.js [hunk 2]
```diff
@@ -32,59 +32,79 @@ var caLogger = require('logops'),
...
-function sendQueryContext(service, servicePath, payload, callback) {
+function sendQueryContext(request, payload, callback) {
+  var headers = {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json',
+    'Fiware-Service': request.headers[caConfig.HEADER.FIWARE_SERVICE],
+    'Fiware-ServicePath': request.headers[caConfig.HEADER.FIWARE_SERVICE_PATH]
+  };
+  if (caHelper.getCorrelator(request)) {
+    headers['Fiware-Correlator'] = caHelper.getCorrelator(request);
```
Comment (AlvaroVega) [inline lib/context_adapter_communication.js:48]: Assignation does the same that previous condition check

File: lib/context_adapter_helper.js [hunk 3]
```diff
@@ -25,48 +25,47 @@
...
-  if (str && str.length === 0) {
-    return hash;
-  }
-  for (i = 0, len = str.length; i < len; i++) {
-    chr = str.charCodeAt(i);
-    hash = ((hash << 5) - hash) + chr;
-    hash |= 0; // Convert to 32bit integer
-  }
-  return hash;
+function getCorrelator(data) {
```
Comment (AlvaroVega) [inline lib/context_adapter_helper.js:36]: If no correlator is found in headers, it will use transaction_id as correlator or it will be always different?

**Comment by AlvaroVega** [comment]
LGTM

File: lib/context_adapter_helper.js (hunk 3 above)
Comment (gtorodelvalle) [inline lib/context_adapter_helper.js:36]: If no correlator is found in headers al the logging messages will appear with `corr=n/a' a new internal transaction id will be generated per request and all the related logging messages will show it in the 'trans' context property.

File: lib/context_adapter_communication.js (hunk 2 above)
Comment (gtorodelvalle) [inline lib/context_adapter_communication.js:48]: Yeah, I just wanted to include the header it was passed in the notification request or include a new one if it was not included.

**Pull request merged**